### PR TITLE
docs: Ensure code is compliant with new github workflows

### PR DESCRIPTION
**Pull request by marwinbaumannsbp**
- Ensure code is compliant with new github workflows
- Restructure codebase to make it easier to find code, put code that relates together

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,28 +5,8 @@
 *.tfstate
 *.tfstate.*
 
-# Crash log files
-crash.log
+# .tfvars files
+*.tfvars
```
Comment (shoekstra) [inline .gitignore:9]: don't think we need this comment? it repeats directly what is beneath

File: main.tf [hunk 2]
```diff
@@ -139,6 +138,33 @@ resource "github_team_repository" "readers" {
   repository = github_repository.default.name
 }
 
+################################################################################
+# Actions
+################################################################################
+resource "github_actions_repository_access_level" "actions_access_level" {
```
Comment (shoekstra) [inline main.tf:144]:
```suggestion

resource "github_actions_repository_access_level" "actions_access_level" {
```

File: examples/main-protected/versions.tf
Comment (shoekstra) [inline examples/main-protected/versions.tf:1]: We don't need this if the same version is in the root module

File: examples/main-unprotected/versions.tf
Comment (shoekstra) [inline examples/main-unprotected/versions.tf:1]: We don't need this if the same version is in the root module

File: examples/master-protected/versions.tf
Comment (shoekstra) [inline examples/master-protected/versions.tf:1]: We don't need this if the same version is in the root module

File: examples/master-unprotected/versions.tf
Comment (shoekstra) [inline examples/master-unprotected/versions.tf:1]: We don't need this if the same version is in the root module

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (marwinbaumannsbp) [reply .gitignore:9]: see slack

**Review by shoekstra — APPROVED** [review]

**Pull request merged**